File: app/src/main/java/com/projectcenterfvt/historicalpenza/Dialogs/LogoutDialog.java
package com.projectcenterfvt.historicalpenza.Dialogs;

import android.app.Dialog;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;

import com.projectcenterfvt.historicalpenza.Activity.SplashActivity;
import com.projectcenterfvt.historicalpenza.Managers.PreferencesManager;
import com.projectcenterfvt.historicalpenza.R;

/**
 * Created by MaksimS on 04.05.2018.
 */

public class LogoutDialog  extends android.support.v4.app.DialogFragment {
    private Dialog dialog;

    @Nullable
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        dialog = super.onCreateDialog(savedInstanceState);
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        Window dialogWindow = dialog.getWindow();
        dialogWindow.setLayout(350, 400);
        dialogWindow.setBackgroundDrawable(new ColorDrawable(0));

        return dialog;
    }
    private void signOut() {
        PreferencesManager preferencesManager = new PreferencesManager(getContext());
        preferencesManager.setToken(" ");
        preferencesManager.setFirstTime(true);

        dialog.dismiss();
        startActivity(new Intent(getActivity(), SplashActivity.class));
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.log_out_menu, null);
        v.setBackgroundResource(R.drawable.dialog_bgn);

        v.findViewById(R.id.btnBackForth).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dialog.dismiss();
            }
        });
        v.findViewById(R.id.button_log_out).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                signOut();
            }
        });
        return v;
    }

}

File: app/src/main/java/com/projectcenterfvt/historicalpenza/Dialogs/LogoutDialog.kt
package com.projectcenterfvt.historicalpenza.dialogs

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.gms.auth.api.signin.GoogleSignIn
import com.google.android.gms.auth.api.signin.GoogleSignInOptions

import com.projectcenterfvt.historicalpenza.sign_in.SignInActivity
import com.projectcenterfvt.historicalpenza.data.Preferences
import com.projectcenterfvt.historicalpenza.R
import com.projectcenterfvt.historicalpenza.data.LandmarksRepository
import kotlinx.android.synthetic.main.dialog_log_out.*
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch


class LogOutDialog : BaseDialog() {

    private val preferences: Preferences by lazy {
        Preferences.getInstance(context!!)
    }

    private val repository: LandmarksRepository by lazy {
        LandmarksRepository.getInstance(context!!)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.dialog_log_out, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        positiveButton.setOnClickListener {
            val signInOptions = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
                    .requestIdToken(getString(R.string.client_server_id))
                    .build()
            val googleSignInClient = GoogleSignIn.getClient(context!!, signInOptions)

            googleSignInClient.signOut().addOnCompleteListener {
                with(preferences) {
                    token = ""
                    shouldNotify = true
                }

                GlobalScope.launch {
                    repository.resetLandmarks()
                }

                val intent = Intent(activity, SignInActivity::class.java).apply {
                    flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
                }
                startActivity(intent)

                dismiss()
                activity?.finish()
            }
        }
        negativeButton.setOnClickListener {
            dismiss()
        }
    }

}
